{{cookiecutter.project_name}}/Modules/Scripted/Home/Home.py

```diff
@@ -89,15 +98,38 @@ def setSlicerUIVisible(self, visible: bool):
         slicer.util.setToolbarsVisible(visible, keepToolbars)
 
     def modifyWindowUI(self):
-        # Custom toolbar
+        """Customize the entire user interface to resemble the custom application"""
+        # Custom toolbars
+        self.initializeAppToolBar()
+        self.initializeSettingsToolBar()
+
+    def initializeAppToolBar(self):
+        """Create parent toolbar for the application"""
+        appToolBar = qt.QToolBar(HomeWidget.APP_TOOLBAR_NAME)
+        appToolBar.name = HomeWidget.APP_TOOLBAR_NAME
+        appToolBar.setMovable(False)
+        appToolBar.setFloatable(False)
         mainToolBar = slicer.util.findChild(slicer.util.mainWindow(), "MainToolBar")
-        self.CustomToolBar = qt.QToolBar("CustomToolBar")
-        self.CustomToolBar.name = "CustomToolBar"
-        slicer.util.mainWindow().insertToolBar(mainToolBar, self.CustomToolBar)
+        slicer.util.mainWindow().insertToolBar(mainToolBar, appToolBar)
+        self._toolbars[appToolBar.name] = appToolBar
+
+    def addToolBar(self, toolBarName: str) -> qt.QToolBar:
+        """Helper method to initialize a new custom toolbar and parent it to the application toolbar"""
```

Comment:
It is not "parented" it is merely inserted before the the named toolbar.  I am not sure that we one to hardcode this, as we maybe be inserting toolbars in various locations for different applications (especially if built in toolbars are used)

Reply:
> appToolBar

This toolbar is intended to include all the toolbars specific to the custom applications, for example:

| |
|--|
| ![image](https://github.com/user-attachments/assets/d823d9f8-35ee-4e3f-a399-82d287b99d2e) |
| ![image](https://github.com/user-attachments/assets/eb4d8311-59eb-4216-8c75-1308aa347f24) |
| ![image](https://github.com/user-attachments/assets/84e1dee3-3795-4209-ae6f-ec3987807d3f) |

Reply:
Supporting inserting toolbar in arbitrary location is indeed sensible. I will revisit the approach.

Reply:
See: https://doc.qt.io/qt-6/qmainwindow.html#insertToolBar

Reply:
Keeping a list of custom toolbars to make show/hide actions easier is a good idea, but the toolbar helper functions just reduce the actual flexibility of toolbar creation

Reply:
> But that's the thing, it does not "include" them. Toolbars don't nest like that. It's just an empty toolbar.

To provide additional context, the idea with the `appToolBar` is support adding  application specific toolbars under only one without having having to use `insertToolBar`.

`insertToolBar` is sensible only if toolbar need to be "injected" within the existing set of Slicer toolbars, which is also a valid use case.

I will then look at supporting the two.

Reply:
After further consideration, I removed the  functions related to `appToolBar` and simplified the API as you suggested.